res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -955,14 +1111,49 @@ PioneerDDJSX.clearHotCueButtons = function(channel, control, value, status, grou
 };
 
 PioneerDDJSX.beatloopButtons = function(channel, control, value, status, group) {
-    var index = control - 0x20,
+    var index = control - 0x50,
         deck = PioneerDDJSX.channelGroups[group];
     script.toggleControl(
         group,
         "beatloop_" + PioneerDDJSX.selectedLoopIntervals[deck][index] + "_toggle"
     );
 };
 
+PioneerDDJSX.slicerButtons = function(channel, control, value, status, group) {
+    var index = control - 0x20,
+        padNum = index % 8,
+        deck = PioneerDDJSX.channelGroups[group],
+        startPos = 0,
+        beatsToJump = 0;
+
+    if (PioneerDDJSX.slicerType[deck] === 1) {
+        PioneerDDJSX.padLedControl(group, PioneerDDJSX.ledGroups.slicer, padNum, false, !value);
+    } else {
+        PioneerDDJSX.padLedControl(group, PioneerDDJSX.ledGroups.slicer, padNum, false, value);
+    }
+    PioneerDDJSX.slicerActive[deck] = value ? true : false;
+    PioneerDDJSX.slicerButton[deck] = index % 8;
+
+    if (value) {
+        startPos = PioneerDDJSX.slicerBeatsPassed[deck];
```

Comment:
I will remove this variable and just use `PioneerDDJSX.slicerBeatsPassed[deck]`.